sab2/Networking/Slider - Material/NetworkService+ActorsService.swift

```diff
@@ -0,0 +1,154 @@
+//  Copyright © 2019 Ibtikar Technologies, Co. Ltd. All rights reserved.
+//
+
+import Foundation
+import Moya
+
+extension NetworkManager {
+    
+    func getSlider_MaterialResponse(pageNumber: Int, completion: @escaping (
+        _ result: Swift.Result<ListApiResponse, NetworkError>,
+        _ statusCode: StatusCode?
+        ) -> Void) {
+        
+        provider.request(MultiTarget(ListService.sliderMaterial(page: pageNumber))) { (result) in
+            switch result {
+            case .success(let response):
+                if (200...299 ~= response.statusCode) {
+                    do {
+                        let result = try JSONDecoder().decode(ListApiResponse.self, from: response.data)
+                        completion(.success(result), response.statusCode)
+                       // print(result)
+                    } catch {
+                        completion(.failure(NetworkError.parseError), response.statusCode)
+                    }
+                } else {
+                    // 300-399 ,400-499
+                    do {
+                        var businessError = try JSONDecoder().decode(NetworkError.self, from: response.data)
+                        businessError.type = .business
+                        completion(.failure(businessError), response.statusCode)
+                    } catch {
+                        completion(.failure(NetworkError.parseError), response.statusCode)
+                    }
+                }
+                
+            case .failure(let error):
+                let customError = NetworkError(error: error)
+                completion(.failure(customError), nil)
+            }
+        }
+        
+    }
+    
+    func getVideosResponse( completion: @escaping (
+        _ result: Swift.Result<VideosImagesApiResponse, NetworkError>,
+        _ statusCode: StatusCode?
+        ) -> Void) {
+        
+        provider.request(MultiTarget(ListService.videos)) { (result) in
+            switch result {
+            case .success(let response):
+                if (200...299 ~= response.statusCode) {
+                    do {
+                        let result = try JSONDecoder().decode(VideosImagesApiResponse.self, from: response.data)
+                        completion(.success(result), response.statusCode)
+                        
+                    } catch {
+                        completion(.failure(NetworkError.parseError), response.statusCode)
+                    }
+                } else {
+                    // 300-399 ,400-499
+                    do {
+                        var businessError = try JSONDecoder().decode(NetworkError.self, from: response.data)
+                        businessError.type = .business
+                        completion(.failure(businessError), response.statusCode)
+                    } catch {
+                        completion(.failure(NetworkError.parseError), response.statusCode)
+                    }
+                }
```

Comment:
Repeating the same bulk of code
create like an ApiClient with only one function of calling network